truss/remote/baseten/remote.py

```diff
@@ -199,6 +203,22 @@ def get_remote_logs_url(
     ) -> str:
         return service.logs_url(self._remote_url)
 
+    async def _watch(self, watch_path: Path, truss_ignore_patterns: List[str]):
+        """Watches for changes in watch_path and sends patch requests for detected changes.
+
+        Changes are detected asynchronously using watchfiles.awatch."""
+
+        def watch_filter(_, path):
+            return not is_ignored(
+                Path(path),
+                truss_ignore_patterns,
+            )
+
+        async for _ in awatch(
+            watch_path, watch_filter=watch_filter, raise_interrupt=False
+        ):
+            self.patch(watch_path)
```

Comment:
maybe for a followup but I wonder if this patch method should also be async since it's essentially just making http calls to the server. That being said there is a limitation where patches where be run in order and only one at a time on the server. I think we should revisit this after the wake investigation gives us a better idea of potential issues on the backend.

Reply:
Agreed. Currently ~80% of the time spent in `patch` is waiting for a response from the server (I'll also update the linear with more results from profiling):
<img width="1366" alt="image" src="https://github.com/basetenlabs/truss/assets/14193683/1e8c0e2d-044a-444e-977a-2c3c795d0ccc">

Prioritizing after wake investigation sounds good. The limitations you mention make sense; we could consider processing these in a background thread but synchronously in the order the patches come in. We would also need to improve the CLI logs to include details on the patch that was created / applied--e.g. an identifier, the changed files--since patch creation logs and "Model patched successfully" logs will be interleaved.

Reply:
Yeah. I also think some kind of batching would make sense. Let's say you change two files then save them one at a time, you likely want both changes to be reflected. Sending those as two separate batches doesn't make sense. We should probably have a 500ms batching window or something

Reply:
Added a ticket: https://linear.app/baseten/issue/BT-9740/improve-performance-of-patch-application-patch-draft-truss